strncpy triggers a build failure in Ubuntu Focal release builds

## Describe the issue

`make install` fails on *Focal* RELEASE builds

https://travis-ci.com/github/astrogeco/cFS/builds/201379042

```
In file included from /usr/include/string.h:495,

                 from /home/travis/build/astrogeco/cFS/cfe/fsw/cfe-core/src/es/cfe_es_apps.c:49:

In function ‘strncpy’,

    inlined from ‘CFE_ES_StartAppTask’ at /home/travis/build/astrogeco/cFS/cfe/fsw/cfe-core/src/es/cfe_es_apps.c:554:9:

/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10: error: ‘__builtin_strncpy’ output may be truncated copying 19 bytes from a string of length 19 [-Werror=stringop-truncation]

  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));

      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

cc1: all warnings being treated as errors
```
_Originally posted by @astrogeco in https://github.com/nasa/cFS/issues/156#issuecomment-727050481_

## Alternative Solution

Use Ubuntu Bionic for builds instead

--- a/fsw/cfe-core/src/es/cfe_es_apps.c
+++ b/fsw/cfe-core/src/es/cfe_es_apps.c
@@ -552,6 +552,7 @@ int32 CFE_ES_StartAppTask(const CFE_ES_AppStartParams_t* StartParams, CFE_ES_Res
 
         TaskRecPtr->AppId = RefAppId;
         strncpy(TaskRecPtr->TaskName, StartParams->BasicInfo.Name, sizeof(TaskRecPtr->TaskName)-1);
+        TaskRecPtr->TaskName[sizeof(TaskRecPtr->TaskName)-1] = 0;
         CFE_ES_TaskRecordSetUsed(TaskRecPtr, TaskId);
 
         /*

